### DISCUSSION TITLE
Using wire:loading on custom component?

### DISCUSSION THREAD

**Discussion by DanjBethel** [livewire/livewire#8472] [1. Help]
wire:loading seems to work fine when I use it on a normal html element. However it has no effect if I use it like this:

<livewire:action-button wire:loading.class="disabled" />

Not sure what I am doing wrong.

**Comment by EdgarsJoja** [comment]
Normally in blade, what you would do is in the component add:
```blade
<div {{ $attributes }}></div>
```
which is known as "attribute forwarding".
But this does not seem to work for Livewire - `Undefined variable $attributes`. No sure if there is any other variable that holds such data.
I suppose the simplest solution would be to wrap your component in `div` and add `wire:loading` for that instead:
```blade
<div wire:loading.class="disabled">
    <livewire:action-button />
</div>
```

**Reply by DanjBethel** [reply]
What happens when I want to show different versions of that same component though? I was hoping to just send a value to my component to change it's state, with your proposed solution it seems like I would have to instead show two different versions of that component?
